server/src/main/java/org/elasticsearch/search/aggregations/metrics/CardinalityAggregatorFactory.java

```diff
@@ -65,10 +84,12 @@ public static void registerAggregators(ValuesSourceRegistry.Builder builder) {
                                 metadata
                             );
                         }
+                        // fallback in the default aggregator
+                        return new KeywordCardinalityAggregator(name, valuesSourceConfig, precision, context, parent, metadata);
```

Comment:
I wonder if `Ordinal` is an important word to have in the name of this thing. Its really like the bytes collecting one but it reasons about ordinals.

Reply:
Yeah, I went back and forth on what to call this.  The next step of this fix is going to be to break this one up, but I didn't want to make that change along with the refactor.  I'll rename it for now, and we can take another pass at it in a few days.